runtime/near-vm-runner/src/metrics.rs

```diff
@@ -19,10 +22,30 @@ pub static COMPILATION_TIME: Lazy<HistogramVec> = Lazy::new(|| {
     .unwrap()
 });
 
+pub static COMPILED_CONTRACT_CACHE_HIT: Lazy<IntCounterVec> = Lazy::new(|| {
+    try_create_int_counter_vec(
+        "near_vm_compiled_contract_cache_hits_total",
+        "The number of times the runtime finds compiled code in cache for the given caller context and shard_id",
+        &["context", "shard_id"],
+    )
+    .unwrap()
+});
+
+pub static COMPILED_CONTRACT_CACHE_MISS: Lazy<IntCounterVec> = Lazy::new(|| {
+    try_create_int_counter_vec(
+        "near_vm_compiled_contract_cache_misses_total",
+        "The number of times the runtime cannot find compiled code in cache for the given caller context and shard_id",
+        &["context", "shard_id"],
+    )
+    .unwrap()
+});
+
 #[derive(Default, Copy, Clone)]
 pub struct Metrics {
     near_vm_compilation_time: Duration,
     wasmtime_compilation_time: Duration,
+    /// True iff the runtime checked the compiled contract cache and found already-compiled code.
+    compiled_contract_cache_hit: Option<bool>,
```

Comment:
I'd use a `usize` that counts the count... It might also solve your concerns with double counting if we make `fn report` operate on the value inside the TLS... So we would change the return value of `run` back to only return an VMResult, and then change `fn report` to be a static method or something that we would operate on the TLS value much like the increment functions do.

Reply:
Yeah I like that. I changed `report` to a static function (renamed to `report_metrics`) and also renamed `reset` to `reset_metrics`. And calling `reset_metrics` and `report_metrics` once in the VM runner (outside of VM.run). PTAL.